### PR TITLE
Management command for auditing API requests

### PR DESCRIPTION
**Pull request by rowanseymour**
Returns status code, time taken, and main index used

screen shot 2016-04-21 at 16 26 13

### CODE REVIEW

File: temba/api/management/commands/apiaudit.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        client.credentials(HTTP_AUTHORIZATION='Token ' + token)
+
+        start_time = time.time()
+        response = client.get(url)
+        time_taken = time.time() - start_time
+
+        # TODO figure out how to get stats to refresh without this workaround.
+        # You'd think that SELECT pg_stat_clear_snapshot() might work but no.
+        connection.close()
+        time.sleep(1)
```
Comment (rowanseymour) [inline temba/api/management/commands/apiaudit.py]: Anyone got any ideas? I tried everything I could think of - calling `VACUUM ANALYZE`, `pg_stat_clear_snapshot()`, putting things in transactions... and the only thing that seems to ensure the `pg_stat_user_indexes` table isn't stale - is closing the db connection AND sleeping for a second.

Comment (nicpottier) [inline temba/api/management/commands/apiaudit.py]: Would be too surprising for pg to only update this on connection closes to make sure it doesn't affect query performance. Certainly a-ok to use that kind of hizhackery for something like this.

**Comment by ericnewcomer** [comment]
neato!

**Comment by nicpottier** [comment]
This would be especially cool if it was something we could run as a separate job against a big db.. maybe by generating fake content.. similarly as the perf tests that were built the real value comes in when we run these automatically and fail on regressions (or on particular thresholds)

Looks good though.

**Comment by nicpottier** [comment]
PS. I fixed travis, so maybe that's the type of thing we could run on our own builder..

**Comment by rowanseymour** [comment]
Yeah that would be nice. Merging this now as is, as it will take some work to setup a suitable test database, but it would be easy to have this return a particular error code that can be incorporated into a Jenkins build.

**Pull request merged**